### ISSUE TITLE
Part.write(String) javadoc does not make sense

### ISSUE THREAD

**Issue by janbartel** [eclipse-ee4j/servlet-api#274]
Despite the modifications to the javadoc in https://github.com/eclipse-ee4j/servlet-api/issues/172, the text still does not make sense for absolute paths:

>       @param fileName The location into which the uploaded part should
>        be stored. The value may be a file name or a path.  The actual
>        location of the file in the filesystem is relative to {@link
>        javax.servlet.MultipartConfigElement#getLocation()}.  Absolute
>        paths are used as provided and are relative to
>        <code>getLocation()</code>.  Note: that this is a system
>        dependent string and URI notation may not be acceptable on all
>        systems. For portability, this string should be generated with
>        the File or Path APIs.

Absolute paths are used as provided, _but_ yet are relative to the configured location?  Does this mean that if an absolute path is provided, it must begin with the same path as the location?  If it does not, should an IOException be thrown?

**Comment by markt-asf** [comment]
I've just re-read the [EG discussion](https://download.oracle.com/javaee-archive/servlet-spec.java.net/jsr369-experts/2017/03/0558.html) on this. I suggest we use my last proposal in that thread (including the modification Greg suggested) of:
```
@param fileName The location into which the uploaded part should be
                stored. Relative paths are relative to {@link
                javax.servlet.MultipartConfigElement#getLocation()}.
                Absolute paths are used as provided.
                Note: that this is a system dependent string and URI
                notation may not be acceptable on all systems. For
                portability, this string should be generated with
                the File or Path APIs.
```

**Comment by gregw** [comment]
@janbartel I'm happy with that text, so can you make a PR?

**Comment by joakime** [comment]
This seems overlapping with Issue #172

**Comment by markt-asf** [comment]
#172 was the original issue associated with the [EG discussion](https://download.oracle.com/javaee-archive/servlet-spec.java.net/jsr369-experts/2017/03/0558.html) I linked above. In short, the change that was made to "fix" #172 wasn't the one that the EG reached consensus on and included the nonsensical phrase called out in this issue. The same concern was raised in the EG thread but was not addressed. Good to see this fixed.
